### PR TITLE
Add LC Capstone Evals

### PR DESCRIPTION
**Pull request by rshipp**
### Description

Modify the "peer review" feature to also work in LC Playbook courses, with minimal changes.

This PR includes only the non-breaking changes. A followup PR will include breaking changes to routes and controllers, which will invalidate currently published "peer review" assignments, as well as #611.

### Motivation and Context

Task: https://app.asana.com/0/1174274412967132/1200131453886866/f

### Screenshots (if appropriate):

Screen Shot 2021-05-24 at 2 51 17 PM

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

File: app/controllers/peer_review_submissions_controller.rb [hunk 1]
```diff
@@ -7,20 +7,45 @@ class PeerReviewSubmissionsController < ApplicationController
...
+    accelerator_canvas_course_id = SalesforceAPI.client.get_accelerator_course_id_from_lc_playbook_course_id(
+      @course.canvas_course_id,
+    )
+
+    if accelerator_canvas_course_id.present?
+      # This is the LC Playbook course.
+      @lc_mode = true
+      # TODO: Dedup this from fellow_evaluation_submissions_controller!
+      # Determine the Accelerator course for this LC Playbook course
+      accelerator_canvas_course_id = SalesforceAPI.client.get_accelerator_course_id_from_lc_playbook_course_id(
```
Comment (sadleb) [inline app/controllers/peer_review_submissions_controller.rb:26]: Accidentally calling the API again?

File: app/controllers/peer_review_submissions_controller.rb [hunk 2]
```diff
@@ -7,20 +7,45 @@ class PeerReviewSubmissionsController < ApplicationController
...
-      # You're not enrolled as a student e.g., you're a TA or admin, or there
-      # are no other students in this course
-      @peer_users = []
+      # This is the Accelerator (Fellow) course.
+      @fellow_mode = true
+      # Get the section in this course that the user is enrolled in as a student
+      student_section = current_user.student_section_by_course(@course)
+      if student_section&.students
+        # You're enrolled as a student and there are other students in this course
+        @eval_users = student_section.students.where.not(id: current_user.id)
```
Comment (sadleb) [inline app/controllers/peer_review_submissions_controller.rb:47]: I assume this doesn’t query twice? If so, does `student_section&.students.where.not` behave the same?

File: app/controllers/peer_review_submissions_controller.rb [hunk 3]
```diff
@@ -31,9 +56,9 @@ def set_questions
   # Called by Submittable.create.
   def answers_params_hash
     peer_review_params = params.require(:peer_review)
-    # Only allow submitting peer reviews for actual peers.
+    # Only allow submitting capstone evals for this Fellow's peers, or this LC's Fellows.
     peer_review_params.permit!.to_h.filter do |user_id|
-      @peer_users.where(id: user_id).exists?
```
Comment (sadleb) [inline app/controllers/peer_review_submissions_controller.rb:36]: Having trouble reading this line and determining if it’s what we want, so I’m just relying on specs here

**Review by sadleb — APPROVED** [review]
The double SF API call is the only thing that is merge blocking (assuming I’m not overlooking something). Approving in advance of that change.

File: app/controllers/peer_review_submissions_controller.rb (hunk 1 above)
Comment (rshipp) [reply app/controllers/peer_review_submissions_controller.rb:26]: yep 🙃

File: app/controllers/peer_review_submissions_controller.rb (hunk 2 above)
Comment (rshipp) [reply app/controllers/peer_review_submissions_controller.rb:47]: It probably does. optimizing with `if &.exists?` seems to work

**Comment by rshipp** [comment]
Merging this without passing CI since the CI is broken 🥴

**Pull request merged**